### ISSUE TITLE
arm-linux-androideabi

### ISSUE THREAD

**Issue by rubdos** [rust-lang-nursery/stdsimd#363]
I'm using stdsimd as an optional dependency for `curve25519-dalek`. Works fine on Intel, but I seem to be having issues compiling for Android.

```
   Compiling stdsimd v0.0.4
error[E0432]: unresolved import `coresimd::__unstable_detect_feature`
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/stdsimd-0.0.4/src/lib.rs:133:9
    |
133 | pub use coresimd::__unstable_detect_feature;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `__unstable_detect_feature` in the root

error: aborting due to previous error
```

I wasn't sure whether this was a known issue, so I'm dropping it off here... I've seen the [issue about the missing CI](https://github.com/rust-lang-nursery/stdsimd/issues/176), but no open issues on Android support.

Easy reproducer:

`cargo build --features=yolocrypto` on https://github.com/dalek-cryptography/curve25519-dalek.git , I'm doing this with Android API 18, NDK r16b.

**Comment by gnzlbg** [comment]
Currently run-time feature detection "should work" on Android as is. It is currently explicitly disabled because of lack of CI support for it. I'll send a PR to enable it once the MIPS PR #365 one is merged, but just because your code might compile does not mean that it will work properly.

Adding CI support for android should not be very hard though. I'll might get to it later today and send a PR with that as well.

**Comment by rubdos** [comment]
> Adding CI support for android should not be very hard though. I'll might get to it later today and send a PR with that as well.

Feel free to tag me in the PR; interested in the follow up.

**Comment by alexcrichton** [comment]
@rubdos this error message is for the `stdsimd` crate on crates.io, but this crate is now available through libstd via `std::arch` and `std::simd`, and I believe the Android target works in there?

**Comment by rubdos** [comment]
> @rubdos this error message is for the stdsimd crate on crates.io, but this crate is now available through libstd via std::arch and std::simd, and I believe the Android target works in there?

Then it's actually curve25519-dalek that's outdated; I'll give it a try. I'll reopen if it's not the case; this sounds very plausible.

**Comment by gnzlbg** [comment]
@rubdos let us know here if it works.

**Comment by rubdos** [comment]
@gnzlbg it seems like the implementation on curve25519-dalek uses pretty old stdsimd code, so I'm pretty certain that that's the case. I might give it a shot in the future, but it's not a priority :-)
